src/test/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/EventGen.scala

```diff
@@ -0,0 +1,351 @@
+      ti_name <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ti_category <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ti_price <- Gen.option(Arbitrary.arbitrary[Double])
+      ti_quantity <- Gen.option(Gen.chooseNum(1, 100))
+      pp_xoffset_min <- Gen.option(Gen.chooseNum(1, 10000))
+      pp_xoffset_max <- Gen.option(Gen.chooseNum(1, 10000))
+      pp_yoffset_min <- Gen.option(Gen.chooseNum(1, 10000))
+      pp_yoffset_max <- Gen.option(Gen.chooseNum(1, 10000))
+      useragent <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_name <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_family <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_version <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_type <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_renderengine <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_lang <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_features_pdf <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_flash <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_java <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_director <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_quicktime <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_realplayer <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_windowsmedia <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_gears <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_features_silverlight <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_cookies <- Gen.option(Arbitrary.arbitrary[Boolean])
+      br_colordepth <- Gen.option(strGen(512, Gen.alphaNumChar))
+      br_viewwidth <- Gen.option(Gen.chooseNum(1, 10000))
+      br_viewheight <- Gen.option(Gen.chooseNum(1, 10000))
+      os_name <- Gen.option(strGen(512, Gen.alphaNumChar))
+      os_family <- Gen.option(strGen(512, Gen.alphaNumChar))
+      os_manufacturer <- Gen.option(strGen(512, Gen.alphaNumChar))
+      os_timezone <- Gen.option(strGen(512, Gen.alphaNumChar))
+      dvce_type <- Gen.option(strGen(512, Gen.alphaNumChar))
+      dvce_ismobile <- Gen.option(Arbitrary.arbitrary[Boolean])
+      dvce_screenwidth <- Gen.option(Gen.chooseNum(1, 10000))
+      dvce_screenheight <- Gen.option(Gen.chooseNum(1, 10000))
+      doc_charset <- Gen.option(strGen(512, Gen.alphaNumChar))
+      doc_width <- Gen.option(Gen.chooseNum(1, 10000))
+      doc_height <- Gen.option(Gen.chooseNum(1, 10000))
+      tr_currency <- Gen.option(strGen(512, Gen.alphaNumChar))
+      tr_total_base <- Gen.option(Arbitrary.arbitrary[Double])
+      tr_tax_base <- Gen.option(Arbitrary.arbitrary[Double])
+      tr_shipping_base <- Gen.option(Arbitrary.arbitrary[Double])
+      ti_currency <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ti_price_base <- Gen.option(Arbitrary.arbitrary[Double])
+      base_currency <- Gen.option(strGen(512, Gen.alphaNumChar))
```

Comment:
This and a few others, like `doc_charset`, `lang` etc, are properly enumerations and not just random strings.